packages/dashboard/src/components/DashboardGridLayout/DashboardGridLayoutWithAddCardPanel/DashboardGridLayoutWithAddCardPanel.tsx

```diff
@@ -222,8 +254,8 @@ export class DashboardGridLayoutWithAddCardPanel extends BaseComponent<
         newLayout.lg!.push(itemLayout);
       });
 
-      // for object comparision using Object.id method
-      if (Object.is(newLayout, this.state.layout)) {
+      // for object comparison using JSON.stringify method
+      if (JSON.stringify(newLayout) === JSON.stringify(this.state.layout)) {
```

Comment:
Rajesh is on vacation, so should be back on tuesday, he will be able to address the comments then. Thanks